omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +274,191 @@ else if(identifierSourceType.equals(Identifier_Pool)){
+            if(identifierSourceToUpdate != null){
+                IdentifierPool pool = (IdentifierPool) identifierSourceToUpdate;
+                Object batchSize = updateBody.get("batchSize");
+                Set<PooledIdentifier> previousIdentifiers = pool.getIdentifiers();
+                if(pool != null && batchSize != null && StringUtils.isNumeric(batchSize.toString())){
+                    Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, Integer.parseInt(batchSize.toString()));
+                }
+                Set<PooledIdentifier> newIdentifiers = pool.getIdentifiers();
+                List<PooledIdentifier> generatedIdentifiers = new ArrayList();
+                for(PooledIdentifier identifier: newIdentifiers){
+                    if(!previousIdentifiers.contains(identifier)){
+                        generatedIdentifiers.add(identifier);
+                    }
+                }
+                return output.add("identifiers", generatedIdentifiers.toString());
+            }
+        }
+        if (operation != null&& operation.toString().equals("uploadFromFile")) {
+            Object identifiers = updateBody.get("identifiers");
+            if(identifierSourceToUpdate != null && identifiers != null){
+                List<String> ids = new ArrayList<String>(Arrays.asList(identifiers.toString().split(",")));
+                IdentifierPool pool = (IdentifierPool) identifierSourceToUpdate;
+                Set<PooledIdentifier> previousIdentifiers = pool.getIdentifiers();
+                if(pool != null){
+                    Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, ids);
+                } 
+                Set<PooledIdentifier> newIdentifiers = pool.getIdentifiers();
+                List<PooledIdentifier> generatedIdentifiers = new ArrayList();
+                for(PooledIdentifier identifier: newIdentifiers){
+                    if(!previousIdentifiers.contains(identifier)){
+                        generatedIdentifiers.add(identifier);
+                    }
+                }
+                return output.add("identifiers", generatedIdentifiers.toString());
+            }
+        }
+        if (generateIdentifiers != null) {
+            Object comment = updateBody.get("comment");
+            Object numberToGenerate = updateBody.get("numberToGenerate");
+            if (comment == null) {
+                comment = "Batch Export of " + numberToGenerate + " to file";
+            }
+            if(numberToGenerate == null){
+                numberToGenerate = 0;
+            }
+            if(identifierSourceToUpdate != null){
+                List<String> batch = Context.getService(IdentifierSourceService.class).
+                    generateIdentifiers(identifierSourceToUpdate, Integer.parseInt(numberToGenerate.toString()), comment.toString());
+                output.add("identifiers", batch.toString()); 
+            }
+            return output;
```

Comment:
Why are you retuning these? Are they for any later use for whoever makes the REST call?

Reply:
Yes. I display the identifiers to the user.

Reply:
That looks like tying the API to a particular user interface. If any particular user interface needs to see what was exactly uploaded, then they need to make a new REST call.

Reply:
@dkayiwa If I stop returning the identifiers uploaded or created, we will get back the test errors (conflict due to isRetired)  because I will have to make GET requests to see if the new identifiers are present.

Reply:
Isn't that like hiding a wound instead of treating it?

Reply:
Fixed. Removed the unnecessary return statements. Only one remains which is required for exporting identifiers.